Gems/AudioSystem/Code/Source/Editor/AudioControl.h

```diff
@@ -25,46 +25,48 @@ namespace AudioControls
     {
         SRawConnectionData(AZ::rapidxml::xml_node<char>* node, bool isValid)
         {
-            m_xmlNode = AZStd::move(DeepCopyNode(node));
+            m_xmlNode = DeepCopyNode(node);
             m_isValid = isValid;
         }
 
-        AZStd::unique_ptr<AZ::rapidxml::xml_node<char>> m_xmlNode{};
+        AZ::rapidxml::xml_node<char>* m_xmlNode = nullptr;
 
         // indicates if the connection is valid for the currently loaded middleware
         bool m_isValid{ false };
 
+    private:
         // Rapid XML provides a 'clone_node' utility that will copy an entire node tree,
         // but it only copies pointers of any strings in the node names and values.
-        // This causes problems with storing raw xml nodes as this class does because strings
-        // will be pointing into the memory pool of an xml document that has gone out of scope.
+        // This causes problems with storage of xml trees, as this class does, because strings
+        // will be pointing into an xml document's file buffer that has gone out of scope.
         // This function is a rewritten version of 'clone_node' that does the deep copy of strings
         // into the new destination tree.
-        [[nodiscard]] static AZStd::unique_ptr<AZ::rapidxml::xml_node<char>> DeepCopyNode(AZ::rapidxml::xml_node<char>* srcNode)
+        [[nodiscard]] AZ::rapidxml::xml_node<char>* DeepCopyNode(AZ::rapidxml::xml_node<char>* srcNode)
         {
-            AZStd::unique_ptr<AZ::rapidxml::xml_node<char>> destNode;
-            if (srcNode)
+            if (!srcNode)
+            {
+                return nullptr;
+            }
+
+            XmlAllocator& xmlAlloc(AudioControls::s_xmlAllocator);
```

Comment:
So none of the memory will be cleaned up until the XmlAllocator goes out of scope?

Reply:
Yeah that's one of the pitfalls of this solution.  It will need a bit of further refactor to get to a more sensible lifetime management.  Discussing with @lawsonamzn about it.